CADETPythonSimulator/solver.py

```diff
@@ -0,0 +1,264 @@
+            Calculated derivatives of y to save
+
+        """
+        it = 0
+        for state in self.unit_solutions.values():
+            for state_dict in state.values():
+                for sol_tuple in state_dict.values():
+                    itp = it + sol_tuple["values"].shape[1]
+
+                    y = y_history[:,it:itp]
+                    ydot = y_dot_history[:,it:itp]
+
+                    sol_tuple["values"] = np.concatenate((sol_tuple["values"], y))
+                    sol_tuple["derivatives"] = np.concatenate((
+                        sol_tuple["derivatives"], ydot
+                    ))
+                    it = itp
+
+        self.time_solutions = np.concatenate((self.time_solutions, times))
+
+    def solve(self) -> NoReturn:
+        """Simulate the system."""
+        self.initialize_system()
+        self.initialize_solution_recorder()
+
+        for section in self.sections:
+            self.solve_section(section)
+
+    def initialize_system(self):
+        """Initialize System."""
+        self.system.initialize()
+
+    def solve_section(
+            self,
+            section: Dict,
+            ) -> NoReturn:
+        """
+        Solve a time section of the differential-algebraic equation system.
+
+        This method uses the solver initialized by the system to compute the states and
+        their derivatives over a specified section.
+
+        Parameters
+        ----------
+        section : Dict
+            The time points at which the solution is sought.
+        #TODO: Consider creating a section class instead of using Addict
+
+        """
+        self._update_unit_operation_parameters(
+            section.start,
+            section.end,
+            section.section_states,
+        )
+        self._system.update_system_connectivity(section.connections)
+
+        section_solution_times = self.get_section_solution_times(section)
+
+        y_initial = self._system.y
+        y_initial_dot = self._system.y_dot
+        output = self.solver.solve(
+            section_solution_times, y_initial, y_initial_dot
+        )
+
+        y_history = output.values.y
+        y_dot_history = output.values.ydot
+        times = output.values.t
+
+        self._system.y = y_history[-1]
+        self._system.y_dot = y_dot_history[-1]
+
+        self.write_solution(times, y_history, y_dot_history)
+
+    def get_section_solution_times(self, section: Dict) -> np.ndarray:
+        """
+        Calculate Section Times.
+
+        Every Section describes the solution times with a start and end time.w
+        The solution times are simply calculated by using np.arange with time_resolution
+        as step size. If time_resolution is not given, a resolution of 1s is used.
+
+        Parameters
+        ----------
+        section : Dict
+            Section Dict that describes the Parameters
+
+        """
+        # TODO: How to get section_solution_times from section.start, section.end, if 
```

Comment:
still some trailing whitespace here and there. :sweat_smile:

Reply:
Between # and todo or what do you mean in this case?